src/main/java/io/vertx/kafka/client/consumer/impl/KafkaReadStreamImpl.java

```diff
@@ -137,24 +138,37 @@ public KafkaReadStreamImpl(Context context, Consumer<K, V> consumer) {
   }
 
   private void pollRecords(Handler<ConsumerRecords<K, V>> handler) {
-    this.worker.submit(() -> {
-      if (!this.closed.get()) {
-        try {
-          ConsumerRecords<K, V> records = this.consumer.poll(pollTimeout);
-          if (records != null && records.count() > 0) {
-            this.context.runOnContext(v -> handler.handle(records));
-          } else {
-            // Don't call pollRecords directly, but use schedule() to actually pause when the readStream is paused
-            schedule(0);
-          }
-        } catch (WakeupException ignore) {
-        } catch (Exception e) {
-          if (exceptionHandler != null) {
-            exceptionHandler.handle(e);
-          }
-        }
+      if(!this.polling.compareAndSet(false, true)){
```

Comment:
I think that we could make this method always run on Event Loop and remove the Atomic here with:

```
private void pollRecords(Handler<ConsumerRecords<K, V>> handler) {
  if (Vertx.currentContext() == this.context) {
    pollRecordsImpl(handler);
  } else {
    vertx.runOnContext(v -> pollRecordsImpl(handler));
  }
}

private void pollRecordsImpl(Handler<ConsumerRecords<K, V>> handler) {
  ...
}
```

this way we avoid the CAS and most of the time this will be anyway called from the event loop.